core/templates/domain/statistics/ExplorationStatsObjectFactory.ts

```diff
@@ -67,6 +69,37 @@ export class ExplorationStats {
     }
     return this.stateStatsMapping.get(stateName);
   }
+
+  withStateAdded(newStateName: string): ExplorationStats {
```

Comment:
Odd naming -- "with" is not a verb. Try reframing so that it is?

Ditto below.